doc/reference/actor.md

```diff
@@ -2,755 +2,738 @@
+default the registry is reachable at `/tmp/ensemble/registry` in the
+current host.
 
-### rpc-register
-::: tip usage
-```
-(rpc-register ...)
+### set-default-registry-addresses!
+```scheme
+(set-default-registry-addresses! addrs)
 ```
-:::
 
-Please document me!
+Sets the default registry addresses.
 
-### rpc-unregister
-::: tip usage
+### server-address-cache-ttl
+```scheme
+(server-address-cache-ttl)
 ```
-(rpc-unregister ...)
-```
-:::
 
-Please document me!
+Returns the actor's server address cache TTL in seconds; by default this is 5 minutes.
 
-### bind-protocol!
-::: tip usage
-```
-(bind-protocol! ...)
+### set-server-address-cache-ttl!
+```scheme
+(set-server-address-cache-ttl! ttl)
 ```
-:::
 
-Please document me!
+Sets the actor server's address cache TTL (in seconds).
 
 
-## RPC Server Protocol
 
-### rpc.register
-::: tip usage
-```
-(rpc.register ...)
-(!rpc.register ...)
-(!!rpc.register ...)
+## Ensemble Servers
+
+### call-with-ensemble-server
+```scheme
+(call-with-ensemble-server server-id thunk
+                           log-level: (log-level 'INFO)
+                           log-file:  (log-file #f)
+                           listen:    (listen-addrs [])
+                           announce:  (public-addrs #f)
+                           registry:  (registry-addrs #f)
+                           roles:     (roles [])
+                           cookie:    (cookie (get-actor-server-cookie)))
 ```
-:::
 
-Please document me!
+This is the programmatic equivalent of `gxensemble run`; first it
+starts the logger with the appropriate options, and then it starts a
+new actor server with identifier `server-id`, starts the loader
+service, adds the server to the ensemble, and then invokes thunk.
+When the thunk exits, it shuts down the actor server and removes the
+server-id from the ensemble.
 
-### rpc.unregister
-::: tip usage
-```
-(rpc.unregister ...)
-(!rpc.unregister ...)
-(!!rpc.unregister ...)
+Options:
+- `log-level`: logging level to use; INFO by default
+- `log-file`: log file for the logger; if it is "-" then the canonical
+  server log is used; this file is at
+  `$GERBIL_PATH/ensemble/server/<server-id>/log`.
+- `listen`: a list of addresses for the actor server to listen to, in
+  addition to the default unix address.
+- `announce`: an optional list of addresses to announce to the registry,
+  in addition to the default unix address. If it is not specified, then
+  the listen addresses are announced.
+- `registry`: an optional list of registry addresses. If it is not specified,
+  then the default registry address is used.
+- `roles`: a list of roles the server fullfills in the registry.
+- `cookie`: the cookie to use; by default it uses the ensemble cooke in
+  `$GERBIL_PATH/ensemble/cookie`.
+
+### ensemble-base-path
+```scheme
+(ensemble-base-path)
 ```
-:::
 
-Please document me!
+Returns the base directory for the ensemble; this is `$GERBIL_PATH/ensemble`.
```

Comment:
Should it be ${XDG_DATA_HOME}/ensemble ? ${XDG_DATA_HOME}/gerbil ?

Reply:
No.